model/project.go

```diff
@@ -681,6 +682,54 @@ func GetTaskGroup(taskGroup string, tc *TaskConfig) (*TaskGroup, error) {
 	return tg, nil
 }
 
+// EarlierInTaskGroup returns true if t1 occurs earlier in the taskGroup, false
+// if t2 occurs earlier. If neither is present in the task group, it returns an
+// error. For efficiency it does not call LoadProjectInto but instead only
+// unmarshals the task group portion of the project's YAML.
+func EarlierInTaskGroup(t1, t2 *task.Task, taskGroup string) (bool, error) {
+	// validate input
+	if taskGroup == "" {
+		return false, errors.New("taskGroup is empty")
+	}
+	if t1.Version != t2.Version || t1.Version == "" || t2.Version == "" {
+		return false, errors.New("versions must match")
+	}
+	if t1.BuildId != t2.BuildId || t1.BuildId == "" || t2.BuildId == "" {
+		return false, errors.New("builds must match")
+	}
+	if t1.TaskGroup != t2.TaskGroup || t1.TaskGroup == "" || t2.TaskGroup == "" {
+		return false, errors.New("task groups must match")
+	}
+
+	// get config
+	v, err := version.FindOneId(t1.Version)
```

Comment:
I worry that we're going to end up slowing the scheduler down with this. 

Theoretically, the second argument to the comand parator can/should cache versions.

The problem is that we'll not know if it matters until we're under production loads, I think.

Reply:
Merging after discussion with a ticket to cache versions.